### PR TITLE
Convert old docstring types to Py3 typehints and run some basic automatic linter checks

### PR DESCRIPTION
**Pull request by Heufneutje**
So I thought it would be nice to catch basic stuff that could go wrong and made pyflakes run automatically when something is pushed or when a PR is made. To do this properly I converted all the old docstring types to Python 3 type hints. I probably broke a few things, hence why I didn't push this right away so this may need some testing.

### CODE REVIEW

File: desertbot/desertbot.py [hunk 1]
```diff
@@ -117,12 +117,12 @@ def irc_JOIN(self, prefix, params):
 
         message = IRCMessage('JOIN', prefix, channel, u'', self)
 
-        if message.User.Name == self.nickname:
```
Comment (rasek-sls) [inline desertbot/desertbot.py]: These changes are not right, it's still User and ReplyTo in the IRCMessage class

Comment (HubbeKing) [inline desertbot/desertbot.py]: These changes are not right, it's still User and ReplyTo in the IRCMessage class

Comment (HubbeKing) [reply desertbot/desertbot.py]: Changes from L120 - L268 in desertbot.py should be reverted back to point at User and ReplyTo for now.

Comment (Heufneutje) [reply desertbot/desertbot.py]: Oops. I actually tried to refactor those but reverted it to not cram too much into one PR. Looks like I didn't revert all of it...

**Comment by TyranicMoron** [comment]
Apart from the bit Hubbe pointed out, the rest of this looks good to me.
Haven't tried running it yet though, that will have to wait till tonight.

**Comment by Heufneutje** [comment]
Alright, `desertbot.py` has been reverted properly and only modifies the type hints now.

File: desertbot/desertbot.py [hunk 2]
```diff
@@ -7,16 +7,16 @@
 
 from twisted.words.protocols import irc
 from twisted.internet import reactor
+from desertbot.config import Config
+from desertbot.factory import DesertBotFactory
```
Comment (TyranicMoron) [inline desertbot/desertbot.py]: This is causing a cyclic import and stopping the code from running.

File: desertbot/message.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
 # -*- coding: utf-8 -*-
 from enum import Enum
+from typing import Dict
 import re
 
+from desertbot.desertbot import DesertBot
```
Comment (TyranicMoron) [inline desertbot/message.py]: And so is this, if I remove the other one. I assume more would show up if I continued.

I didn't know python 3's type hinting was implemented this badly =/

**Comment by TyranicMoron** [comment]
Ah, I see. Couple more and I can get the bot to launch.

File: desertbot/modulehandler.py [hunk 4]
```diff
@@ -9,18 +9,17 @@
 from twisted.internet import threads
 from enum import Enum
 from six import iteritems
+from typing import Any, List
 
 from desertbot.moduleinterface import IModule
 import desertbot.modules
-from desertbot.message import TargetTypes
+from desertbot.desertbot import DesertBot
```
Comment (TyranicMoron) [inline desertbot/modulehandler.py]: Here

File: desertbot/moduleinterface.py [hunk 5]
```diff
@@ -3,11 +3,15 @@
 from zope.interface import Interface
 from functools import wraps
 from fnmatch import fnmatch
+from typing import Any, Callable, List, Tuple
 import logging
 
+from desertbot.desertbot import DesertBot
```
Comment (TyranicMoron) [inline desertbot/moduleinterface.py]: And here.

**Comment by Heufneutje** [comment]
Try again (also looks like Travis is actually doing its thing now).

**Pull request merged**